src/main/java/de/tum/in/www1/artemis/repository/GradingScaleRepository.java

```diff
@@ -52,9 +54,20 @@
      */
     @NotNull
     default GradingScale findByCourseIdOrElseThrow(Long courseId) {
-        return findByCourseId(courseId).orElseThrow(() -> new EntityNotFoundException("Grading scale with course ID " + courseId + " doesn't exist"));
+        try {
+            return findByCourseId(courseId).orElseThrow(() -> new EntityNotFoundException("Grading scale with course ID " + courseId + " doesn't exist"));
+        }
+        catch (IncorrectResultSizeDataAccessException exception) {
+            List<GradingScale> gradingScales = findAllByCourseId(courseId);
```

Comment:
Can you please add some documentation why this code is necessary.
Do you have a server integration tests that checks if it works?

Reply:
There is an integration test. It is necessary because of some concurrency issues. Since only one grading scale is allowed for course/exam if for some reason more than one was saved, the grading scales could be modified or retrieved (the IncorrectResultSizeDataAccessException occurred). That caused some problems if someone decided to spam the save button for example - the server can already handle post requests if a grading scale exists, as long it has already been saved to the database. If, however, the endpoint was entered multiple times before an actual grading scale got saved, multiple entries would be persisted, and then every subsequent query would result in this exception and not retrieve a grading scale at all. I've added some handling on the client to disable the save/delete buttons if the grading scale is being retrieved/saved (the isLoading property) to prevent that from happening again, but this code change will
1) fix any potential duplicated entries on the test servers the first time someone queries the grading scale in some way.
2) act as a fail-safe if for some reason multiple entries still manage to get saved.